keps/3589-manage-jobs-selectively/README.md

```diff
@@ -0,0 +1,233 @@
+	// unsuspended, they will start immediately.
+	ManageJobsWithoutQueueName bool `json:"manageJobsWithoutQueueName"`
+
+  // ManageAllJobsNamespaceSelector can be used to omit some namespaces from ManageJobsWithoutQueueName
+	ManagedJobsNamespaceSelector *metav1.LabelSelector `json:"managedJobsNamespaceSelector,omitempty"`
+
+  ...
+}
+```
+
+### Implementation
+
+The configuration of all Job webhooks are extended to propagate the namespace selector and it is
+used to filter `manageJobsWithoutQueueName` as is currently done in the Pod integration. Specifically,
+jobs without queue names are only managed if both `manageJobsWithoutQueueName` is true and
+the jobs namespace matches the `managedJobsNamespaceSelector`.
+
+Configuration parsing during startup is extended to give a deprecation notice if `podOptions.namespaceSelector` is set.
+Setting both `podOptions.namespaceSelector` and `managedJobsNamespaceSelector` will be flagged as a configuration error.
+
+### Test Plan
+
+[ X ] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+#### Unit Tests
+
+Unit tests for configuration parsing will be added.
+
+All unit tests for the current Pod integration namespaceSelector will be extended across all integrations.
+
+#### Integration tests
+
+All integration tests for the current Pod integration namespaceSelector will be extended across all integrations.
+
+## Implementation History
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
+
+### Do nothing
+
+We could do nothing and simply document that despite its historical placement in
+a subfield of the `Configuration` struct, `podOptions.namespaceSelector` also applies
+to the `StatefulSet` and `Deployment` integrations.
+
+### Validating Admission Policies
+
+As previously discussed in [issue 2119](https://github.com/kubernetes-sigs/kueue/issues/2119),
+we could phase out `manageJobsWithoutQueueNames` and instead recommend the
+use of ValidatingAdmissionPolicies to ensure that all Kinds that could be managed by Kueue that
+are created in "user" namespaces have a queue name label (and thus will be managed by Kueue).
+
+However, while exploring this idea further a complication arose. The ValidatingAdmissionPolicy
+must be able to distinguish between top-level resources without queue names (which should be denied)
+and child resources that are owned by Kueue-managed parents (which must be admitted even if they do not have a queue name).
+In other words, the VAP must implement the same functionality as `jobframework.IsOwnerManagedByKueue`
+and cross check controller-references against managed kinds.  This approach may be challenging to scale
+as the number of Kueue managed kinds increases and ancestor trees become deeper.
+For example in the not so distant future we might see: AppWrapper->PyTorchJob->JobSet->Job->Pod.
```

Comment:
Wouldn't it be enough to do the same as is done in `jobframework.IsOwnerManagedByKueue` and just check if the parent is manageable by Kueue? So ancestor tree size wouldn't matter.

Reply:
I will try to phrase this better.  It isn't that the check has to actually crawl up the ancestor tree.  It's that the check has to consider each of the kinds that could be Kueue-managed parents.  So it needs a longer list of kinds and it has to somehow know that a particular Kind's Kueue integration is enabled.   For kinds like Job or JobSet that could have multiple Kueue-managed parent Kinds, each has to be covered by the check.